packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -139,9 +139,9 @@ export const errorNames = {
   manifestTemplateInvalid: "ManifestTemplateInvalid",
   aadManifestTemplateNotExist: "AadManifestTemplateNotExist",
 };
-const upgradeButton = getLocalizedString("core.option.upgrade");
-export const moreInfoButton = getLocalizedString("core.option.moreInfo");
-const migrationMessageButtons = [upgradeButton, moreInfoButton];
+const upgradeButton = () => getLocalizedString("core.option.upgrade");
```

Comment:
Overall looks good to me, can you share some background about why we need to make it a function?

Reply:
The `locale` is passed when running, so use `const string` will always use `locale=en` and cannot modify when `locale` has been changed.